techmatters/hrm
# Add extra functionality to the existing permissions framework

## Description
This PR adds two new type of conditions that can be used in our permissions framework, `createdHoursAgo` and `createdDaysAgo`. This are slightly different that the conditions we had so far, since they are not plain strings but objects that require a number in conjunction to the condition. E.g. 
- `{ "createdHoursAgo": 24 }` will be true if the target was created less than 24 hours ago.
  - `[["isSupervisor"], ["isOwner", { "createdHoursAgo": 24 }]]` allows access if the user is a supervisor or if the user is the owner of a contact and the contact was created less than 24 hours ago.
- `{ "createdDaysAgo": 2 }` will be true if the target was created less than 2 days ago.
  - `[["isSupervisor"], ["isOwner", { "createdDaysAgo": 2 }]]` allows access if the user is a supervisor or if the user is the owner of a contact and the contact was created less than 2 days ago.


### Checklist
- [x] [Corresponding issue has been opened](https://tech-matters.atlassian.net/browse/CHI-2362)

### Verification steps
- Trust tests or
- Change development permissions rules to use the new predicates, as indicated above and confirm that the actions are evaluated accordingly.
